go/obscuronode/enclave/sql/sqlite.go

```diff
@@ -3,43 +3,73 @@ package sql
 import (
 	"database/sql"
 	"fmt"
+	"math/rand"
 	"os"
 	"path/filepath"
+	"time"
+
+	"github.com/obscuronet/obscuro-playground/go/log"
 
 	"github.com/ethereum/go-ethereum/ethdb"
 	_ "github.com/mattn/go-sqlite3" // this imports the sqlite driver to make the sql.Open() connection work
 )
 
 const (
-	tempDirName = "temp-obscuro-persistence"
+	tempDirName = "obscuro-persistence"
 	createQry   = `create table if not exists kv (key binary(32) primary key, value blob); delete from kv;`
 )
 
-func CreateTemporarySQLiteDB(nodeID uint64) (ethdb.Database, error) {
-	dbFile, err := getTempDBFile(nodeID)
-	if err != nil {
-		return nil, fmt.Errorf("failed to create temp sqlite DB file - %w", err)
+// CreateTemporarySQLiteDB takes a filepath which can be empty but allows for tests that care about persistence between restarts
+func CreateTemporarySQLiteDB(dbPath string) (ethdb.Database, error) {
+	if dbPath == "" {
+		tempPath, err := getTempDBFile()
+		if err != nil {
+			return nil, fmt.Errorf("failed to create temp sqlite DB file - %w", err)
+		}
+		dbPath = tempPath
 	}
-	db, err := sql.Open("sqlite3", dbFile)
-	if err != nil {
-		return nil, fmt.Errorf("couldn't create temp sqlite db - %w", err)
+	// determine if a db file already exists, we don't want to overwrite it
+	_, err := os.Stat(dbPath)
+	newOrExisting := "new"
```

Comment:
Yeah it was part of this PR unfortunately haha. Been fighting Go pretty hard on this, it's exactly the situation you want a ternary condition isn't it.

Had it as a boolean and set the string in the if statement where I initialise a new DB and then the linter started failing because I was only using the boolean variable in a single if statement! I'll just do two if statements, screw you Go :)

Reply:
Better just to suppress the linter, no?